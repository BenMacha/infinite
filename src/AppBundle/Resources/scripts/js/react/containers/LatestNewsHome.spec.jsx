import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
// eslint-disable-next-line import/no-extraneous-dependencies

import LatestNewsHome from './LatestNewsHome.jsx';

jest.mock('actions', () => ({
  latestNews: {
    fetch: jest.fn(),
    prefetch: jest.fn(),
    refresh: jest.fn(),
  },
}));

describe('LatestNews container', () => {
  const header = {
    text: 'Lâ€™info en continu',
    tag: 'h1',
  };

  it('render correctly', () => {
    // Mocker Modules
    // eslint-disable-next-line global-require
    const actions = require('actions');

    const latestNews = {
      data: [
        {
          id: 42,
          publication_url: '/path/to/article/42',
          publish_date: '2016-08-31T12:54:52.956Z',
          content: {
            restricted: false,
            title: 'Article without type',
            section: 'pixels',
          },
        },
        {
          id: 42,
          publication_url: '/path/to/article/42',
          publish_date: '2016-08-31T12:54:52.956Z',
          content: {
            restricted: false,
            title: 'The article invisible',
            section: 'mlemag',
          },
          type: 'article',
          visible: false,
        },
        {
          id: 42,
          publication_url: '/path/to/article/42',
          publish_date: '2016-08-31T12:54:52.956Z',
          content: {
            restricted: false,
            title: 'The article title',
            section: 'pixels',
          },
          type: 'article',
          visible: true,
        },
        {
          id: 42,
          publish_date: '2016-09-06T23:06:23.281Z',
          publication_url: '/path/to/article/42',
          content: {
            text: 'The article text',
            title: 'The article title',
          },
          type: 'alert',
          visible: true,
        },
      ],
      options: {
        header,
        gtmId: 'latest-news',
        qaId: 'latest-news',
      },
    };
    const store = {
      dispatch: jest.fn(),
      getState: jest.fn(() => ({ latestNews })),
      subscribe: jest.fn(),
    };

    actions.latestNews.fetch.mockImplementationOnce(() => ({
      data: 'fetch',
    }));
    actions.latestNews.prefetch.mockImplementationOnce(() => ({
      data: 'prefetch',
    }));

    const tree = renderer.create(<LatestNewsHome store={store} />);

    const button = tree.toJSON().children[1].children[0];
    // Simulate click on button
    button.props.onClick();

    expect(tree).toMatchSnapshot();

    expect(store.dispatch.mock.calls[0]).toEqual([
      { data: 'fetch' },
    ]);
    expect(store.dispatch.mock.calls[1]).toEqual([
      { data: 'prefetch' },
    ]);
  });

  it('hide the more button while fetching data', () => {
    const latestNews = {
      data: [],
      options: {
        header,
        gtmId: 'latest-news',
        qaId: 'latest-news',
      },
      isFetching: true,
    };

    const store = {
      dispatch: jest.fn(),
      getState: jest.fn(() => ({ latestNews })),
      subscribe: jest.fn(),
    };
    const tree = renderer.create(<LatestNewsHome store={store} />);

    expect(tree).toMatchSnapshot();
  });
});
