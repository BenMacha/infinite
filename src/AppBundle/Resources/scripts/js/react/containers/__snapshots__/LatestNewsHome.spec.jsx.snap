exports[`LatestNews container hide the more button while fetching data 1`] = `
<div>
  <ul
    className="cm-section-thread__section-list" />
</div>
`;

exports[`LatestNews container render correctly 1`] = `
<div>
  <ul
    className="cm-section-thread__section-list">
    <li
      className="cm-section-thread__article" />
    <li
      className="cm-section-thread__article" />
    <li
      className="cm-section-thread__article" />
  </ul>
  <div
    className="ol-grid__cell ol-grid__cell--full@to-phone-l">
    <a
      aria-label="Afficher les contenus plus anciens"
      className="ca-btn ca-btn--important"
      onClick={[Function]}>
      Voir Plus
    </a>
  </div>
</div>
`;
